### PR TITLE
Add Course creation plugin

### LINKED ISSUE #279: Create plugin to create courses
**Is your feature request related to a problem? Please describe.**
Currently, to create a course a superadmin has to set an ID for the new course, this is very random and does not have like a guide about how to create a correct ID for a new course.

**Describe the solution you'd like**
Delete the current way to create a course, create a new plugin that adds an option that opens a modal window where there are some specific fields related to the new course such as name, group, year and semester. Taking these fields, the ID can be created automatically.

**Describe alternatives you've considered (if applicable)**
For future process, we will modify this plugin to be able to clone courses, including the tasks.

### PR DESCRIPTION
**Pull request by crdgonzalezca**
# Description

Add a new plugin that adds a modal for superadmins to create courses in an easier way. Also, allow the superadmin to select a course to copy tasks from this course.

Fixes #279 

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Locally, taking into account edge cases and possible errors.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: inginious/frontend/plugins/course_creation/static/js/course_creation.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    });
+}
+
+function blurCourseCreationModal() {
+    $("#create_course_modal input").prop("disabled", true);
+    $("#list_copy_courses").prop("disabled", true);
+    $("#submit_create_course").prop("disabled", true);
+}
+
+function unblurCourseCreationModal() {
```
Comment (Heldeg) [inline inginious/frontend/plugins/course_creation/static/js/course_creation.js]: meaby `focusCourseCreationModal()` ?

File: inginious/frontend/plugins/course_creation/static/js/course_creation.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+jQuery(document).ready(function () {
+    onCloseModal();
+    $("#create_course_form").submit(function (e) {
+        e.preventDefault();
+    });
+
+    $("#course_year").val(new Date().getFullYear());
+    $("#course_year").datetimepicker();
```
Comment (Heldeg) [inline inginious/frontend/plugins/course_creation/static/js/course_creation.js]: You could use a const to not duplicated jQuery selector

File: inginious/frontend/plugins/course_creation/static/js/course_creation.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    $("#create_course_modal").on("hidden.bs.modal", function () {
+        $("#create_course_modal input").val("");
+        $("#list_copy_courses").val("-1");
+        $("#create_course_alert").prop("hidden", true);
+    });
+}
+
+jQuery(document).ready(function () {
+    onCloseModal();
+    $("#create_course_form").submit(function (e) {
```
Comment (Heldeg) [inline inginious/frontend/plugins/course_creation/static/js/course_creation.js]: My IDE tells me that `submit` function is deprecated and use `on` or `trigger` function  instead. Maybe is good idea implement that change. I think in this case you should use `on`.

**Review by Heldeg — COMMENTED** [review]
It looks good. I checked it by myself and I didn't find any evident bug. I just leave some comments.

File: inginious/frontend/plugins/course_creation/api/create_course_api.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        if not course_year.isnumeric():
+            raise api.APIError(400, _("The year must be a number."))
+
+        if not course_period:
+            raise api.APIError(400, _("The period field cannot be empty."))
+
+        data = {"name": course_name, "group": course_group, "year": course_year, "period": course_period}
+
+        all_courses = set(self.course_factory.get_all_courses().keys())
+        if course_to_copy_id and course_to_copy_id != "-1":
```
[the remark this answers is not in the document]
Comment (crdgonzalezca) [reply inginious/frontend/plugins/course_creation/api/create_course_api.py]: That value is the default value for the course selector in the modal. I have added a constant with that value to make it more understandable.

**Review by Heldeg — APPROVED** [review]

**Pull request merged**